08_files_lab/02.py

```diff
@@ -1,7 +1,63 @@
-"""
-Write a program that takes 3 names. The first two
-are existing file names and the last is a new file name
-Program should write the lines from the first two 
-files interwinded into the output file
-"""
-
+#!/usr/bin/env python
+"""
+This program writh lines from fine_1 and file_2 alternately into file_2
+"""
+__author__ = 'Eabel'
+
+import sys
+import os
+import argparse
+import itertools
+
+def is_valid_file_path(file_path):
+    if os.path.isfile(file_path):
+        return True
+    else :      
+        print "Error - invalid file path",file_path
+        return False
+
+def create_dictionaries(file_path):
+    if not os.path.exists(os.path.dirname(file_path)):
+        try:
+            os.makedirs(os.path.dirname(file_path))
+        except OSError as exc: 
+            if exc.errno != errno.EEXIST:           
+                return False
+    return True
+
+########################################################################################################################
+#
+#
+def main():
+    parser = argparse.ArgumentParser(description='This program writh lines from fine_1 and file_2 alternately into file_2')
+    
+    parser.add_argument("file_1", help="path to file")
+    parser.add_argument("file_2", help="path to file")
+    parser.add_argument("file_3", help="path to file")
+    args = parser.parse_args()
+    if not is_valid_file_path(args.file_1) or not is_valid_file_path(args.file_2):
+        sys.exit()
+    if os.path.isfile(args.file_3):
+        print "Error - file",args.file_3,"alrady exist"
+        sys.exit()
+    if not create_dictionaries(args.file_3):

```

Comment:
Don't turn your exceptions to error codes - use them all the way. Consider just the line `os.makedirs(os.path.dirname(file_path))` inside a try/except block. You'll get the same result but without defining a function and without the if, resulting in much shorter and cleaner code